matchms/similarity/ModifiedCosine.py

```diff
@@ -0,0 +1,102 @@
+import numba
+import numpy
+from matchms.typing import SpectrumType
+
+
+class ModifiedCosine:
+    """Calculate modified cosine score between spectra.
```

Comment:
Can you explain what the modified cosine similarity is or add a link where it is described.

Reply:
I added a bit more extensive explanation and a references for further details.

Reply:
nice